packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -204,13 +207,29 @@ class Doctor {
         // Future returned by the asyncGuard() is not awaited, we pass an
         // onError callback to it and translate errors into ValidationResults.
         asyncGuard<ValidationResult>(
-          validator.validate,
+          () {
+            final Completer<ValidationResult> timeoutCompleter = Completer<ValidationResult>();
+            final Timer timer = Timer(doctorDuration, () {
+              timeoutCompleter.completeError(
+                Exception('${validator.title} exceeded maximum allowed duration of $doctorDuration'),
+              );
+            });
+            final Future<ValidationResult> validatorFuture = validator.validate();
+            return Future.any<ValidationResult>(<Future<ValidationResult>>[
+              validatorFuture,
+              // This future can only complete with a ToolExit.
```

Comment:
Looks like it just completes with a regular Exception rather then a ToolExit?

Reply:
good catch

Reply:
updated comment

Reply:
(the exception type doesn't really matter, we catch them all on line 227, all that matters is the message)